pennylane/labs/resource_estimation/resources_base.py

```diff
@@ -0,0 +1,384 @@
+
+    **Example**
+
+    The resources can be accessed as class attributes. Additionally, the :class:`~.Resources`
+    instance can be nicely displayed in the console.
+
+    >>> h = re.resource_rep(re.ResourceHadamard)
+    >>> x = re.resource_rep(re.ResourceX)
+    >>> y = re.resource_rep(re.ResourceY)
+    >>>
+    >>> qm = re.QubitManager(work_wires=3)
+    >>> gt = defaultdict(int, {h: 10, x:7, y:3})
+    >>>
+    >>> res = re.Resources(qubit_manager=qm, gate_types=gt)
+    >>> print(res)
+    --- Resources: ---
+    Total qubits: 3
+    Total gates : 20
+    Qubit breakdown:
+    clean qubits: 3, dirty qubits: 0, algorithmic qubits: 0
+    Gate breakdown:
+    {'Hadamard': 10, 'X': 7, 'Y': 3}
+
+    .. details::
+        :title: Usage Details
+
+        The :class:`Resources` object supports arithmetic operations which allow for quick addition
+        and multiplication of resources.
+
+        .. code-block::
+
+            from collections import defaultdict
+
+            # Resource reps for each operator:
+            h = re.resource_rep(re.ResourceHadamard)
+            x = re.resource_rep(re.ResourceX)
+            z = re.resource_rep(re.ResourceZ)
+            cnot = re.resource_rep(re.ResourceCNOT)
+
+            # state of qubits:
+            qm1 = re.QubitManager(work_wires={"clean":2, "dirty":1})
+            qm1.algo_qubits = 3
+
+            qm2 = re.QubitManager(work_wires={"clean":1, "dirty":2})
+            qm2.algo_qubits = 4
+
+            # state of gates:
+            gt1 = defaultdict(int, {h: 10, x:5, cnot:2})
+            gt2 = defaultdict(int, {h: 15, z:5, cnot:4})
+
+            # resources:
+            res1 = re.Resources(qubit_manager=qm1, gate_types=gt1)
+            res2 = re.Resources(qubit_manager=qm2, gate_types=gt2)
+
+
+        .. code-block:: pycon
+
+            >>> print(res1)
+            --- Resources: ---
+            Total qubits: 6
+            Total gates : 17
+            Qubit breakdown:
+            clean qubits: 2, dirty qubits: 1, algorithmic qubits: 3
+            Gate breakdown:
+            {'Hadamard': 10, 'X': 5, 'CNOT': 2}
+
+            >>> print(res2)
+            --- Resources: ---
+            Total qubits: 7
+            Total gates : 24
+            Qubit breakdown:
+            clean qubits: 1, dirty qubits: 2, algorithmic qubits: 4
+            Gate breakdown:
+            {'Hadamard': 15, 'Z': 5, 'CNOT': 4}
+
+        Specifically, users can add together two instances of resources using the :code:`+` and
+        :code:`&` operators. These represent combining the resources assuming the circuits were
+        executed in series or parallel respectively.
+
+        .. code-block:: pycon
+
+            >>> res_in_series = res1 + res2
+            >>> print(res_in_series)
+            --- Resources: ---
+            Total qubits: 9
+            Total gates : 41
+            Qubit breakdown:
+            clean qubits: 2, dirty qubits: 3, algorithmic qubits: 4
```

Comment:
How are the work wires in the series resource blocks interacting? I understood the `algorithmic qubits` but couldn't outright infer the other two.

Reply:
Added a paragraph explaining this.